pennylane/vqe/vqe.py

```diff
@@ -129,6 +130,70 @@ def __str__(self):
 
         return "\n+ ".join(terms)
 
+    def __matmul__(self, H):
+
+        coeffs1 = self.coeffs.copy()
+        terms1 = self.ops.copy()
+
+        coeffs2 = H.coeffs
+        terms2 = H.ops
+
+        coeffs = [i[0] * i[1] for i in itertools.product(coeffs1, coeffs2)]
+        term_list = itertools.product(terms1, terms2)
+
+        terms = [qml.operation.Tensor(i[0], i[1]) for i in term_list]
+
+        return qml.Hamiltonian(coeffs, terms)
+
+    def __add__(self, H):
+
+        coeffs = self.coeffs.copy()
+        ops = self.ops.copy()
+
+        op_attributes = []
+
+        for op in ops:
+            name = op.name if isinstance(op.name, list) else [op.name]
+
+            if "Hermitian" not in name:
+                op_attributes.append(set(zip(name, op.wires)))
```

Comment:
I'm a bit confused. What are these lines doing?

Reply:
Same in line 167. Is this to deal with the case where the operator is not Pauli but a Hermitian operator?

Reply:
Yep, definitely worth commenting below.